packages/langium/src/references/scope-computation.ts

```diff
@@ -122,6 +123,19 @@ export class DefaultScopeComputation implements ScopeComputation {
         return scopes;
     }
 
+    /**
+     * @returns A new precomputed scopes container instance for the given document.
+     */
+    protected newPrecomputedScopes(_document: LangiumDocument): PrecomputedScopes {
+        const map = new MultiMap<AstNode, AstNodeDescription>();
+        return {
+            add:       map.add.bind(map),
+            addAll:    map.addAll.bind(map),
+            get:       map.get.bind(map),
+            getStream: (key: AstNode) => stream(map.get(key))
+        };
+    }
```

Comment:
Suggestion: I don't think that the "default implementation" of `PrecomputedScopes` should be exclusively available as a method inside of the default implementation of `ScopeComputation`. Instead, I would much rather have a `MultiMapPrecomputedScopes` class, that can be used by adopters that don't want to override `DefaultScopeComputation`.